### PR TITLE
Fix potential crash when listing leaked objects

### PR DESCRIPTION
**Pull request by RandomShaper**
See the comment in the code for details.

This is the version of #38507 for 4.0.

---
**This code is generously donated by IMVU.**

### CODE REVIEW

File: core/object.cpp [hunk 1]
```diff
@@ -2123,21 +2123,38 @@ void ObjectDB::cleanup() {
 	if (slot_count > 0) {
 		spin_lock.lock();
 
-		WARN_PRINT("ObjectDB Instances still exist!");
 		if (OS::get_singleton()->is_stdout_verbose()) {
+			WARN_PRINT("ObjectDB Instances still exist!");
+
+			// Ensure calling the native classes because if a leaked instance
+			// has a script overriding nay of those methods, it'd not be OK to
```
Comment (akien-mga) [inline core/object.cpp:2130]: Typo: "nay".
You can also add a dot at the end of the sentence.

File: core/object.cpp [hunk 2]
```diff
@@ -2123,21 +2123,38 @@ void ObjectDB::cleanup() {
 	if (slot_count > 0) {
 		spin_lock.lock();
 
-		WARN_PRINT("ObjectDB Instances still exist!");
```
Comment (akien-mga) [inline core/object.cpp:2126]: I'd take the opportunity to make the "Instances" lower case (same in the other warning).

**Review by akien-mga — APPROVED** [review]
Looks good, just suggested some style changes.

File: core/object.cpp (hunk 2 above)
Comment (RandomShaper) [reply core/object.cpp:2126]: Good point. Fixed. I've also inverted the `if` so it's clearer that both messages need to be updated if one is changed. I thought about doing it in a more elegant way, but it didn't pay.

File: core/object.cpp (hunk 1 above)
Comment (RandomShaper) [reply core/object.cpp]: Both fixed.

**Pull request merged**